src/main/java/edu/cornell/kfs/module/purap/batch/dataaccess/impl/JaggaerUploadDaoJdbc.java

```diff
@@ -100,92 +106,100 @@ public List<JaggaerContractAddressUploadDto> findJaggaerContractAddress(JaggaerC
+                CuSqlChunk.forStringParameters(addressTypes),
+                ") ",
+                buildOrderByClauseCuSqlChunk());
+        return chunk;
     }
     
-    private String buildFromClause(boolean includeVendorAddress) {
-        StringBuilder sb = new StringBuilder("FROM KFS.PUR_VNDR_HDR_T VH, KFS.PUR_VNDR_DTL_T VD");
+    private CuSqlChunk buildFromClauseCuSqlChunk(boolean includeVendorAddress) {
+        CuSqlChunk chunk = CuSqlChunk.of("FROM KFS.PUR_VNDR_HDR_T VH, KFS.PUR_VNDR_DTL_T VD ");
         if (includeVendorAddress) {
-            sb.append(", KFS.PUR_VNDR_ADDR_T VA");
+            chunk.append(", KFS.PUR_VNDR_ADDR_T VA ");
         }
-        sb.append(StringUtils.SPACE);
-        return sb.toString();
+        return chunk;
     }
     
-    private String buildJoinClauseWithPOVendors(boolean includeVendorAddress) {
-        StringBuilder sb = new StringBuilder("WHERE VH.VNDR_HDR_GNRTD_ID = VD.VNDR_HDR_GNRTD_ID ");
+    private CuSqlChunk buildJoinClauseWithPOVendorsCuSqlChunk(boolean includeVendorAddress) {
+        CuSqlChunk chunk = CuSqlChunk.of("WHERE VH.VNDR_HDR_GNRTD_ID = VD.VNDR_HDR_GNRTD_ID ");
         if (includeVendorAddress) {
-            sb.append("AND VA.VNDR_HDR_GNRTD_ID = VD.VNDR_HDR_GNRTD_ID ");
-            sb.append("AND VA.VNDR_DTL_ASND_ID = VD.VNDR_DTL_ASND_ID ");
+            chunk.append("AND VA.VNDR_HDR_GNRTD_ID = VD.VNDR_HDR_GNRTD_ID ");
+            chunk.append("AND VA.VNDR_DTL_ASND_ID = VD.VNDR_DTL_ASND_ID ");
         }
-        sb.append("AND VH.VNDR_TYP_CD = 'PO' ");
-        return sb.toString();
+        chunk.append("AND VH.VNDR_TYP_CD =");
+        chunk.append(CuSqlChunk.forParameter(CUPurapConstants.JaggaerAddressType.FULFILLMENT.kfsAddressTypeCode));
+        chunk.append(StringUtils.SPACE);
+        return chunk;
     }
     
-    private String buildRestrictActiveRows(boolean includeVendorAddress) {
-        StringBuilder sb = new StringBuilder("AND VD.DOBJ_MAINT_CD_ACTV_IND = 'Y' ");
+    private CuSqlChunk buildRestrictActiveRowsCuSqlChunk(boolean includeVendorAddress) {
+        CuSqlChunk chunk = CuSqlChunk.of("AND VD.DOBJ_MAINT_CD_ACTV_IND = ", 
+                CuSqlChunk.forParameter(KFSConstants.ACTIVE_INDICATOR), 
+                StringUtils.SPACE);
         if (includeVendorAddress) {
-            sb.append("AND VA.DOBJ_MAINT_CD_ACTV_IND = 'Y' ");
+            chunk.append("AND VA.DOBJ_MAINT_CD_ACTV_IND = ");
+            chunk.append(CuSqlChunk.forParameter(KFSConstants.ACTIVE_INDICATOR));
+            chunk.append(StringUtils.SPACE);
         }
-        return sb.toString();
+        return chunk;
     }
     
-    private String buildTimeFrameRestrictionClause(JaggaerContractUploadProcessingMode processingMode, String processingDate) {
-        StringBuilder sb = new StringBuilder();
+    private CuSqlChunk buildTimeFrameRestrictionClauseCuSqlChunk(JaggaerContractUploadProcessingMode processingMode, String processingDate) {
```

Comment:
Now that you're using the parameterized setup, I would suggest using a SQL Date or SQL Timestamp for the processingDate, either parsed way in advance or parsed within the DAO class. (The same goes for the related methods in this class.) That way, you don't have to include the TO_DATE() function within the SQL.

If you think such a change would be too impacting at this time, then I would recommend creating a follow-up user story to make that change.

Reply:
I coverted